web/src/lib/roadmap/roadmapBuilder.test.ts

```diff
@@ -84,9 +82,9 @@ describe("roadmapBuilder", () => {
       ])
     );
 
-    expect(roadmap.steps[1].tasks.map((it) => it.id)).toEqual(expect.arrayContaining(["coffee-task-3-id"]));
+    expect(roadmap.tasks.map((it) => it.id)).toEqual(expect.arrayContaining(["coffee-task-3-id"]));
 
-    expect(roadmap.steps[2].tasks.map((it) => it.id)).toEqual(expect.arrayContaining(["coffee-task-4-id"]));
+    expect(roadmap.tasks.map((it) => it.id)).toEqual(expect.arrayContaining(["coffee-task-4-id"]));
```

Comment:
so with these changes here, we're actually losing a piece of confidence that we used to have while testing.

In the previous version, the tests were asserting _not just_ that the tasks were showing up in the roadmap but _also_ that the step numbers were being interpreted correctly. 

We were previously testing that step 1 contained certain tasks, and step 2 contained certain tasks, and so on. I think in order to keep the same degree of confidence now, we need to test that the step numbers are being used correctly in the new structure.

I have a suggestion on how to do this, let me know if you want me to share, but otherwise wanted to give you a chance to think through it on your own

Reply:
this is true for pretty much all the test changes in this file

Reply:
I will try to take a stab at it right now.